packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -2988,14 +2988,31 @@ export class ThinEngine extends AbstractEngine {
             samplingMode,
             onLoad,
             onError,
-            this._prepareWebGLTexture.bind(this),
+            (
+                texture: InternalTexture,
+                extension: string,
+                scene: Nullable<ISceneLike>,
+                img: HTMLImageElement | ImageBitmap | { width: number; height: number },
+                invertY: boolean,
+                noMipmap: boolean,
+                isCompressed: boolean,
+                processFunction: (
+                    width: number,
+                    height: number,
+                    img: HTMLImageElement | ImageBitmap | { width: number; height: number },
+                    extension: string,
+                    texture: InternalTexture,
+                    continuationCallback: () => void
+                ) => boolean,
+                samplingMode: number

```

Comment:
Having to copy/paste all this is hard to read and hard to maintain. If we are ok with this change, then I would also propose factoring this out into an exported type from AbstractEngine. Any objections?

Reply:
As it is in thinEngine it should not affect the size so no problem for me
@sebavan ?

Reply:
no problem either, it is only typings